lib/list_versions.go

```diff
@@ -126,17 +109,71 @@ func GetTFURLBody(mirrorURL string) ([]string, error) {
 		os.Exit(1)
 	}
 
-	body, errBody := ioutil.ReadAll(resp.Body)
-	if errBody != nil {
-		log.Printf("[Error] : reading body: %v", errBody)
+	body := new(bytes.Buffer)
+	if _, err := io.Copy(body, resp.Body); err != nil {
+		log.Fatal(err)
+	}
+	if err := json.Unmarshal(body.Bytes(), &releases); err != nil {
+		log.Fatalf("%s: %s", err, body.String())
+	}
+	return releases, nil
+}
+
+func GetTFReleases(mirrorURL string, preRelease bool) ([]Release, error) {
+	// temp testing
+	queryParams := map[string]string{"limit": "20"}
+	rel, _ := getReleases(mirrorURL, queryParams)
+
+	var releases []Release
+	rel, _ = getReleases(mirrorURL, queryParams)
+	releases = append(releases, rel...)
+	for len(rel) == 20 {
```

Comment:
Argh, `20` once again — putting it as a variable indeed makes sense imho.

Reply:
Also the question since I'm not any good with Go: why `for` and not `if`?

Reply:
Because here I'm achieving a while loop vs a single conditional compare. I'm intentionally continuing the loop until the list is smaller than `limit`, which is the point we know we're done.

Reply:
Got you. Thanks for the eplanation.